### ISSUE TITLE
[testnet] After deleting and re-syncing blockchain, ownership of an asset seemed to be lost when trying to transfer while it's not really lost 

### ISSUE THREAD

**Issue by bedri** [RavenProject/Ravencoin#224]
### Describe the issue
When the blockchain deleted for some reason (I deleted it after -reindex core dump problem) and start the wallet for re-syncing, while asset creation and transfer blockchain data is restored and asset holding addresses' ownership remains, both raven-qt and raven-cli refuses to show that the assets are owned by the addresses that the wallet has the private keys of.

```
raven-cli -testnet listmyassets
{
  "COM/DOMAIN/WWW": 1
}
```
> listmyassets does not list LINK asset which is actually owned by the address that this wallet has the private key of.

```
raven-cli -testnet listmyassets LINK
{
  "LINK": 2000
}
```
> But "listmyassets LINK" command correctly shows the owned asset

```
raven-cli -testnet listaddressesbyasset LINK
{
  "mhwpH3m3zE9AKyJTokJZLyoVUkNNPkauCM": 1000,
  "msC1E5XGa2jZYr6H6YBQDXSNDEndC6F89H": 1000
}
```
> These two addresses holds LINK asset

```
raven-cli -testnet listaddressgroupings | grep msC1E5XGa2jZYr6H6YBQDXSNDEndC6F89H
      "msC1E5XGa2jZYr6H6YBQDXSNDEndC6F89H",
```
> This address is owned by this wallet

```
raven-cli -testnet listaddressgroupings | grep mhwpH3m3zE9AKyJTokJZLyoVUkNNPkauCM
      "mhwpH3m3zE9AKyJTokJZLyoVUkNNPkauCM",
```
> This address is also owned by this wallet

```
raven-cli -testnet transfer LINK 1 mwcQ1kQfWqMW2STKb8S34zsotkhweccBZ7
error code: -32602
error message:
This wallet doesn't own any assets with the name: LINK
```
> But still when trying to transfer some amount from this asset, it's gives error and refuses that this wallet owns this asset.

raven-qt also does not list the owned assets in the Overview tab under "Asset Balances" part
![raven-2 0 3-asset-test-7](https://user-images.githubusercontent.com/765487/44126011-6cd71844-a03e-11e8-8489-8f725a0f2840.png)

Again the asset's issue transaction and an reissue transaction is in the wallet's transaction list
![raven-2 0 3-asset-test-8](https://user-images.githubusercontent.com/765487/44126112-ff603d3a-a03e-11e8-8069-def04611e103.png)

#### If so, please list the steps to reproduce below:
1. Start raven-qt with the "raven-qt -testnet -daemon -server" command
2. Issue an asset
3. Reissue the asset if needed
4. Stop raven-qt
5. Delete .raven/testnet4/blocks directory
6. Start raven-qt with the "raven-qt -testnet -daemon -server" command again
7. Observe missing assets in the Overview -> "Asset Balances" box
8. Check transactions and observe the issue and reissue transactions are still in the list
9. Try to reissue or transfer the asset and observe that it's not in the assets list
10. Use the commands in the "### Describe the issue" above and observe the problem exists as it's told.

### Expected behaviour
Owned asset should be listed in the "raven-qt -> Overview -> Asset Balances" as well as in "raven-cli listmyassets" command's output.
Asset should able to be transfered without an error.

### Actual behaviour
While the wallet keeping private keys of the addresses that holds the asset actually issued and reissued the asset and owns it, it cannot transfer the asset and "listmyassets" rpc command does not list the asset.

### Screenshots.
Screenshots given above.

**Comment by blondfrogs** [comment]
@bedri If you are deleting your blocks folder you must also delete your assets folder. This will allow the chain to update your asset database as it gets blocks.

**Comment by bedri** [comment]
@blondfrogs I did that and the asset database did not rebuilt from scratch.  -rescan also did not bring back the assets that are invisible.

I had backed up assets directory earlier. I had copied it to testnet4 directory and the lost asset came back but this time the recent assets created after the backup are missing.

Since -reindex is giving segmentation fault on my desktop system, i will try -reindex on ubuntu 16.04 to check if asset database will be built from scratch with reindexing.

If I can trace the error I will also try to fix it and create a pull request.

Thank you for your attention.
